pkg/ccl/changefeedccl/parquet_sink_cloudstorage.go

```diff
@@ -0,0 +1,254 @@
+// existing separation between encoder and the sync, then we would need to
+// figure out a way to get the embedded parquet writer in the
+// cloudStorageSinkFile and pass it to the encode function in the encoder.
+// Instead of this it logically made sense to have a single sink for parquet
+// format which did the job of both encoding and emitting to the cloud storage.
+// This sink currently embeds the cloudStorageSinkFile and it has a function
+// EncodeAndEmitRow which links the buffer in the cloudStorageSinkFile and the
+// parquetWriter every time we need to create a file for each unique combination
+// of topic and schema version (We need to have a unique parquetWriter for each
+// unique combination of topic and schema version because each parquet file can
+// have a single schema written inside it and each parquetWriter can only be
+// associated with a single schema.)
+type parquetCloudStorageSink struct {
+	wrapped *cloudStorageSink
+}
+
+type parquetWriterWrapper struct {
+	parquetWriter  *goparquet.FileWriter
+	schema         *parquetschema.SchemaDefinition
+	parquetColumns []pqexporter.ParquetColumn
+	numCols        int
+}
+
+func makeParquetCloudStorageSink(baseCloudStorageSink *cloudStorageSink) *parquetCloudStorageSink {
+	pqcs := &parquetCloudStorageSink{}
+
+	pqcs.wrapped = baseCloudStorageSink
+
+	return pqcs
+}
+
+// EmitRow does not do anything. It must not be called. It is present so that
+// parquetCloudStorageSink implements the Sink interface.
+func (pqcs *parquetCloudStorageSink) EmitRow(
+	ctx context.Context,
+	topic TopicDescriptor,
+	key, value []byte,
+	updated, mvcc hlc.Timestamp,
+	alloc kvevent.Alloc,
+) error {
+	return errors.Errorf("Emit Row should not be called for parquet format")
+}
+
+// Close implements the Sink interface.
+func (pqcs *parquetCloudStorageSink) Close() error {
+	return pqcs.wrapped.Close()
+}
+
+// Dial implements the Sink interface.
+func (pqcs *parquetCloudStorageSink) Dial() error {
+	return pqcs.wrapped.Dial()
+}
+
+// EmitResolvedTimestamp does not do anything as of now. It is there to
+// implement Sink interface.
+func (pqcs *parquetCloudStorageSink) EmitResolvedTimestamp(
+	ctx context.Context, encoder Encoder, resolved hlc.Timestamp,
+) error {
+	return errors.Errorf("parquet format does not support emitting resolved timestamp")
+}
+
+// Flush implements the Sink interface.
+func (pqcs *parquetCloudStorageSink) Flush(ctx context.Context) error {
+	return pqcs.wrapped.Flush(ctx)
+}
+
+// EncodeAndEmitRow links the buffer in the cloud storage sync file and the
+// parquet writer (see parquetCloudStorageSink). It also takes care of encoding
+// and emitting row event to cloud storage. Implements the SinkWithEncoder
+// interface.
+func (pqcs *parquetCloudStorageSink) EncodeAndEmitRow(
+	ctx context.Context,
+	updatedRow cdcevent.Row,
+	topic TopicDescriptor,
+	updated, mvcc hlc.Timestamp,
+	alloc kvevent.Alloc,
+) error {
+	s := pqcs.wrapped
+	var pqww *parquetWriterWrapper
```

Comment:
nit: In general, acronyms are not suitable variable names. variable names should give the reader a good hint of its purpose. Call pqww `fileWriter` instead?

Reply:
also, you could replace `pqcs` with `parquetSink`. Though it seems more conventional in go to abbreviate a function's struct.